sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DataTypeParserSuite.scala

```diff
@@ -61,6 +61,7 @@ class DataTypeParserSuite extends SparkFunSuite {
   checkDataType("varchAr(20)", StringType)
   checkDataType("cHaR(27)", StringType)
   checkDataType("BINARY", BinaryType)
+  checkDataType("void", NullType)
```

Comment:
Could you add end-2-end tests, too, just like this?
```
scala> sql("create table t (v void)")
scala> sql("insert into t values (null), (null), (null)")
scala> sql("select * from t").show()
// Checks result output
```

Reply:
The result output above is `Row(null) :: Row(null) :: Row(null) :: Nil`.
But I remember we shouldn't allow `create table t (v void)` https://github.com/apache/spark/pull/17953#issuecomment-306056712. So it needs additional PR to throw exception. #25198 is the PR which to do that.

Reply:
If so, could we fix the issue, too, in this PR?

Reply:
Sure, I will complete it.

Reply:
Thanks!

Reply:
After reviewed #25198, I find @ulysses-you had performed much work for that. @ulysses-you do you want to continue to complete that?

Reply:
I don't know what is needed, in other word I don't know what is needed to be compatible with Hive. It may be the reason why this thing was left. 

There are 2 different things between Spark and Hive.
1. Hive support both table and view with `Void` type, like `create table t as select null as c`, `create view v as select null as c`. Spark doesn't support both.
2. For null value type, Hive use `Void` and Spark use `NullType`

As @cloud-fan said, we shouldn't support `NullType` for tables. But may compatible with Hive in some DDLs like `desc table`, `show create talbe` ?

I think we should discuss what things to be compatible with Hive first and then move.